test/client.test.js

```diff
@@ -111,6 +111,18 @@ describe('Client', () => {
 
     return expect(resultWithoutOptions).toEqual(resultWithOptions)
   })
+
+  test('uses custom fetch', async function() {
+    const fetch = jest.fn(() =>
+      Promise.resolve({
+        headers: new Set(),
```

Comment:
I believe we want a test that exercises headers as an `Iterator` as is the case when using browser `fetch`

Reply:
The `Iterator` is returned by the `entries()`. The `headers` options are type of `Headers[]` which have a similar API with the `Headers` object. I used `Set()` to avoid extra mocking.